cmd/qmstr-reporter-html/main.go

```diff
@@ -0,0 +1,206 @@
+//go:generate protoc -I ../../pkg/buildservice --go_out=plugins=grpc:../../pkg/buildservice ../../pkg/buildservice/buildservice.proto
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"path"
+	"regexp"
+	"strings"
+	"text/template"
+
+	version "github.com/hashicorp/go-version"
+)
+
+//"github.com/endocode/qmstr/pkg/reporter/htmlreporter"
+
+const minimumHugoVersion = "0.32"
+const themeDirectory = "/usr/share/qmstr/reporter-html/theme"
+
+func main() {
+	detectedVersion, err := DetectHugoAndVerifyVersion()
+	if err != nil {
+		log.Fatalf("error generating HTML reports: %v", err)
+	}
+	log.Printf("detected beautiful Hugo version %v", detectedVersion)
+
+	// htmlreporter.ConnectToMaster()
+	wd, err := CreateHugoWorkingDirectory()
+	if err != nil {
+		log.Fatalf("error preparing temporary Hugo working directory: %v", err)
+	}
+	log.Printf("created temporary Hugo working directory in %v", wd)
+	defer func() {
+		//TODO is there a --keep option or environment variable to keep temporary files?
+		log.Printf("deleting temporary Hugo working directory in %v", wd)
+		if err := os.RemoveAll(wd); err != nil {
+			log.Printf("warning - error deleting temporary Hugo working directory in %v: %v", wd, err)
+		}
+	}()
+	// generate the content data (markdown and JSON) from the master data model
+	// TODO
+
+	staticHTMLContentDir, err := CreateStaticHTML(wd)
+	if err != nil {
+		log.Fatalf("error generating reports: %v", err)
+	}
+	log.Printf("QMSTR reports generated in %v", staticHTMLContentDir)
+
+	// Create the reports package (for publishing, etc).
+	// ... TODO: configure default target directory for every reporter (uses CWD at the moment)
+	packagePath, _ := os.Getwd()
+	if err := CreateReportsPackage(wd, staticHTMLContentDir, packagePath); err != nil {
+		log.Fatalf("error packaging report to %v: %v", packagePath, err)
+	}
+	log.Printf("QMSTR reports package generated in %v", packagePath)
+
+	// defer htmlreporter.DisconnectFromMaster()
+}
+
+// DetectHugoAndVerifyVersion runs Hugo to get the version string.
+func DetectHugoAndVerifyVersion() (string, error) {
+	cmd := exec.Command("hugo", "version", "--quiet")
```

Comment:
Isn't there a way to do this programatically? This program is just calling lots of commands rather than using libraries. This introduces dependencies like hugo to be installed in the master server .